### PR TITLE
Remove validation to stat call for symlinks since is a breaking change

### LINKED ISSUE #57221: Microsoft.Extensions.FileProviders.Physical.Tests crashing on CI in FileSystemInfo API
Build: https://dev.azure.com/dnceng/public/_build/results?buildId=1288068&view=ms.vss-test-web.build-test-results-tab&runId=38025328&resultId=194160&paneView=dotnet-dnceng.dnceng-build-release-tasks.helix-test-information-tab

Configuration: `net6.0-OSX-Debug-x64-CoreCLR_release-OSX.1015.Amd64.Open`

Dump is available: https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-56974-merge-9155b3b7ae8e44c498/Microsoft.Extensions.FileProviders.Physical.Tests/1/how-to-debug-dump.md?sv=2019-07-07&se=2021-08-31T15%3A39%3A48Z&sr=c&sp=rl&sig=Z1XxeIsGVGHUGWG6t0bfozd05lMZIHaZ%2BpmuqDJbn08%3D, https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-56974-merge-9155b3b7ae8e44c498/Microsoft.Extensions.FileProviders.Physical.Tests/1/core.22560?sv=2019-07-07&se=2021-08-31T15%3A39%3A48Z&sr=c&sp=rl&sig=Z1XxeIsGVGHUGWG6t0bfozd05lMZIHaZ%2BpmuqDJbn08%3D

```
/tmp/helix/working/A95708FE/p/dotnet exec --runtimeconfig Microsoft.Extensions.FileProviders.Physical.Tests.runtimeconfig.json --depsfile Microsoft.Extensions.FileProviders.Physical.Tests.deps.json xunit.console.dll Microsoft.Extensions.FileProviders.Physical.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=IgnoreForCI -notrait category=OuterLoop -notrait category=failing 
popd
===========================================================================================================
/private/tmp/helix/working/A95708FE/w/97EC089D/e /private/tmp/helix/working/A95708FE/w/97EC089D/e
  Discovering: Microsoft.Extensions.FileProviders.Physical.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  Microsoft.Extensions.FileProviders.Physical.Tests (found 76 of 92 test cases)
  Starting:    Microsoft.Extensions.FileProviders.Physical.Tests (parallel test collections = on, max threads = 6)
Unhandled exception. System.IO.IOException: Invalid argument
   at System.IO.FileStatus.ThrowOnCacheInitializationError(ReadOnlySpan`1 path) in /_/src/libraries/System.Private.CoreLib/src/System/IO/FileStatus.Unix.cs:line 421
   at System.IO.FileStatus.EnsureCachesInitialized(ReadOnlySpan`1 path, Boolean continueOnError) in /_/src/libraries/System.Private.CoreLib/src/System/IO/FileStatus.Unix.cs:line 400
   at System.IO.FileStatus.GetLastWriteTime(ReadOnlySpan`1 path, Boolean continueOnError) in /_/src/libraries/System.Private.CoreLib/src/System/IO/FileStatus.Unix.cs:line 284
   at System.IO.FileSystemInfo.get_LastWriteTimeCore() in /_/src/libraries/System.Private.CoreLib/src/System/IO/FileSystemInfo.Unix.cs:line 60
   at System.IO.FileSystemInfo.get_LastWriteTimeUtc() in /_/src/libraries/System.Private.CoreLib/src/System/IO/FileSystemInfo.cs:line 115
   at Microsoft.Extensions.FileProviders.Physical.PollingFileChangeToken.GetLastWriteTimeUtc() in /_/src/libraries/Microsoft.Extensions.FileProviders.Physical/src/PollingFileChangeToken.cs:line 57
   at Microsoft.Extensions.FileProviders.Physical.PollingFileChangeToken.get_HasChanged() in /_/src/libraries/Microsoft.Extensions.FileProviders.Physical/src/PollingFileChangeToken.cs:line 101
   at Microsoft.Extensions.FileProviders.Physical.PhysicalFilesWatcher.RaiseChangeEvents(Object state) in /_/src/libraries/Microsoft.Extensions.FileProviders.Physical/src/PhysicalFilesWatcher.cs:line 455
   at System.Threading.TimerQueueTimer.Fire(Boolean isThreadPool) in /_/src/libraries/System.Private.CoreLib/src/System/Threading/Timer.cs:line 673
   at System.Threading.TimerQueue.FireNextTimers() in /_/src/libraries/System.Private.CoreLib/src/System/Threading/Timer.cs:line 326
./RunTests.sh: line 162: 22560 Abort trap: 6           (core dumped) "$RUNTIME_PATH/dotnet" exec --runtimeconfig Microsoft.Extensions.FileProviders.Physical.Tests.runtimeconfig.json --depsfile Microsoft.Extensions.FileProviders.Physical.Tests.deps.json xunit.console.dll Microsoft.Extensions.FileProviders.Physical.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=IgnoreForCI -notrait category=OuterLoop -notrait category=failing $RSP_FILE
/private/tmp/helix/working/A95708FE/w/97EC089D/e
----- end Wed Aug 11 08:42:00 PDT 2021 ----- exit code 134 ----------------------------------------------------------
exit code 134 means SIGABRT Abort. Managed or native assert, or runtime check such as heap corruption, caused call to abort(). Core dumped.
```

cc @adamsitnik

### PR DESCRIPTION
**Pull request by jozkee**
Subsequently, remove the symlink cache logic as is no longer needed.
Fixes https://github.com/dotnet/runtime/issues/57221

cc @danmoseley

### CODE REVIEW

**Review by carlossanlop — APPROVED** [review]
LGTM. Thanks for investigating this.

File: src/libraries/System.IO.FileSystem/tests/DirectoryInfo/SymbolicLinks.cs [hunk 1]
```diff
@@ -52,72 +52,6 @@ protected override void AssertLinkExists(FileSystemInfo link)
             }
         }
 
-        [Theory]
```
Comment (Jozkee) [inline src/libraries/System.IO.FileSystem/tests/DirectoryInfo/SymbolicLinks.cs:55]: These tests were validating that we throw when a self-referencing symlink is found in the recursion, while that is true for windows, that wasn't the case for Unix in 5.0, and it was going to be a breaking change caused by the same validation that I'm removing, so I updated the tests to ensure what is expected matches with 5.0

**Review by carlossanlop — APPROVED** [review]
LGTM with the latest test updates.

**Comment by Jozkee** [comment]
CI issues are nuget erros and test failure is https://github.com/dotnet/runtime/issues/57452.
Merging now to include it in RC1.

**Pull request merged**